pkg/jmxfetch/jmxfetch_nix.go

```diff
@@ -0,0 +1,58 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build jmx
+// +build !windows
+
+package jmxfetch
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func (j *JMXFetch) Monitor() {
+	idx := 0
+	maxRestarts := config.Datadog.GetInt("jmx_max_restarts")
+	ival := float64(config.Datadog.GetInt("jmx_restart_interval"))
+	stopTimes := make([]time.Time, maxRestarts)
+	ticker := time.NewTicker(500 * time.Millisecond)
+
+	defer ticker.Stop()
+	defer close(j.stopped)
+
+	go j.heartbeat(ticker)
+
+	for {
+		err := j.Wait()
+		if err == nil {
+			log.Infof("JMXFetch stopped and exited sanely.")
+			break
+		}
+
+		stopTimes[idx] = time.Now()
+		oldestIdx := (idx + maxRestarts + 1) % maxRestarts
```

Comment:
This will point to unset stopTimes for the first `maxRestarts` runs. This will work since go initialize them to the zero value of `time.Time`, `0001-01-01 00:00:00 +0000 UTC`. 
Looks a bit confusing. Don't you think it would be better to not check for the first `maxRestarts` runs ?

Reply:
I was counting on the zero value for `time.Time`, hence the implementation. I try to avoid one-off pieces of code when I can, but if you really find this confusing I can change it.

Reply:
I think adding a comment explaining that we rely on this is worth it here